### DISCUSSION TITLE
Physics in the simulation is malfunctioning

### DISCUSSION THREAD

**Discussion by HimGautam** [cyberbotics/webots#4284] [Q&A]
Hi,
I am trying to create a custom quadruped robot by my own. I attached a rectangle to the endpoint of hingejoint. I have used same shape in boundingboxobject and also used the physics but somehow when I run the simulation the legs of robot sink beyond the floor. How can I fix it? Below is the video of the simulation.

https://user-images.githubusercontent.com/70597091/155842753-8a140efd-4980-440d-b546-8edbb675a5a3.mp4

**Comment by ad-daniel** [comment]
Make sure all solids that comprise the robot have physics enabled. Also, have you tried lowering the timestep of the simulation?

**Reply by HimGautam** [reply]
Hi,
Problem was solved by restarting the webots. Now when I run the simulation, several limbs of the robot seems to have moved away from the joint. The video below shows the problem. This problem was seen on premade robots also like GhostDog from EPFL. What can be done to solve this problem?

https://user-images.githubusercontent.com/70597091/155872669-c2c33f48-9c57-41da-8a9d-4232946cce19.mp4
